packages/x-date-pickers/src/locales/ar.ts

```diff
@@ -0,0 +1,97 @@
+import { PickersLocaleText } from './utils/pickersLocaleTextApi';
+import { getPickersLocalization } from './utils/getPickersLocalization';
+import { TimeViewWithMeridiem } from '../internals/models';
+
+// maps TimeView to its translation
+const views: Record<TimeViewWithMeridiem, string> = {
+  hours: 'ساعات',
+  minutes: 'دقائق',
+  seconds: 'ثواني',
+  meridiem: 'ميريديم',
+};
+
+const arPickers: Partial<PickersLocaleText> = {
+  // Calendar navigation
+  previousMonth: 'الشهر السابق',
+  nextMonth: 'الشهر المقبل',
+
+  // View navigation
+  openPreviousView: 'افتح العرض السابق',
+  openNextView: 'افتح العرض التالي',
+  calendarViewSwitchingButtonAriaLabel: (view) =>
+    view === 'year'
+      ? 'عرض السنة مفتوح، قم بالتبديل إلى عرض التقويم'
+      : 'عرض التقويم مفتوح، قم بالتبديل إلى عرض السنة',
+
+  // DateRange labels
+  start: 'يبدأ',
+  end: 'نهاية',
+  startDate: 'تاريخ البدء',
+  startTime: 'وقت البدء',
+  endDate: 'تاريخ الانتهاء',
+  endTime: 'وقت الانتهاء',
+
+  // Action bar
+  cancelButtonLabel: 'يلغي',
+  clearButtonLabel: 'واضح',
+  okButtonLabel: 'نعم',
```

Comment:
We cannot say "Yes" for "ok", because it's usually used in Yes/No confirmations.
```suggestion
  okButtonLabel: 'تم',
```